onnx/reference/reference_evaluator.py

```diff
@@ -541,3 +556,68 @@ def run(self, output_names, feed_inputs: Dict[str, Any], attributes: Optional[Di
                 )
             list_results.append(results[name])
         return list_results
+
+    def _save_intermerdiate_results(
+        self,
+        index: int,
+        node: NodeProto,
+        inputs: List[np.array],
+        outputs: List[np.array],
+    ) -> str:
+        """
+        Saves intermediate results into a folder with the same organization as the backend tests.
+
+        :param index: index of the node in the graph
+        :param node: node proto
+        :param inputs: inputs of the node
+        :param outputs: outputs of the node, supposedly the expected outputs
+        :return: test folder
+        """
+
+        def get_shape(t):
+            return list(t.dims)
+
+        input_protos = [
+            from_array_extended(inp, name) for name, inp in zip(node.input, inputs)
```

Comment:
There are some challenges with control-flow constructs. The above is not going to work for If/Loop since the graphs may refer to other names defined outside (these are implicit inputs). 

A second related question is, of course, what we want to do for control-flow constructs. Do we want one folder for the outer control-flow-op, or sub-folders for the nodes inside (which may execute multiple times for loops)?

It is okay to postpone this to another PR also, since it is non-trivial. Just bringing it up for discussion.

Reply:
done